flytekit/models/security.py

```diff
@@ -42,8 +42,12 @@ class MountType(Enum):
 
     def __post_init__(self):
         from flytekit.configuration.plugin import get_plugin
+        from flytekit.core.context_manager import FlyteContextManager
 
-        if get_plugin().secret_requires_group() and self.group is None:
+        # Only check for the groups during registration.
+        execution = FlyteContextManager.current_context().execution_state
+        in_registration_context = execution.mode is None
```

Comment:
Should we have this to be until function which can be used at other places if needed to determine if we are in registration context

Reply:
If this comes up more often, I would add a new `is_registration` method to `ExecutionState`:

https://github.com/flyteorg/flytekit/blob/afa1fc7f271e042b3f80d6349abc529a13898c84/flytekit/core/context_manager.py#L453

As for this PR, I rather not add public API for now. This is first time in the codebase, we check for `execution.mode is None`.